src/completions.jl

```diff
@@ -164,16 +170,18 @@ completiontype(line, c, mod) = begin
     end
     return found ? completiontype(val, mod, ct) : "ignored"
   end
-  c isa REPLCompletions.KeywordCompletion ? "keyword" :
-    c isa REPLCompletions.PathCompletion ? "path" :
-    c isa REPLCompletions.PackageCompletion ? "import" :
-    c isa REPLCompletions.PropertyCompletion ? "property" :
-    c isa REPLCompletions.FieldCompletion ? "attribute" :
-    c isa REPLCompletions.MethodCompletion ? "method" :
-    "variable"
-end
-completiontype(val, mod::Module, ct::AbstractString) = wstype(mod, Symbol(ct), val)
-completiontype(line, ::REPLCompletions.DictCompletion, mod) = "key" # fallen into "macro" otherwise
+  completiontype(c)
+end
+
+completiontype(line, ::REPLCompletions.DictCompletion, mod) = "key" # DictCompletion isn't dispatched for the entry method, otherwise fallen into "macro"
+completiontype(@nospecialize(val), mod::Module, ct::String) = wstype(mod, Symbol(ct), val) # ModuleCompletion
+completiontype(::REPLCompletions.Completion) = "variable" # may not happen
+completiontype(::REPLCompletions.PackageCompletion) = "import"
+completiontype(::REPLCompletions.MethodCompletion) = "method"
+completiontype(::REPLCompletions.PropertyCompletion) = "property"
+completiontype(::REPLCompletions.FieldCompletion) = "property"
```

Comment:
What `FieldCompletion` does is essentially completing properties of a predicted return type, thus I want to change its type into `property`, and keep `attribute` for the future scope-based completions [described here](https://github.com/JunoLab/Atom.jl/blob/master/src/completions.jl#L34) (, which I would like to tackle in the next JSOC or some other place)

Reply:
Sure, I don't particularly care about the exact symbol we show :)